pkg/settlement/pseudosettle/pseudosettle.go

```diff
@@ -272,7 +272,7 @@ func (s *Service) Pay(ctx context.Context, peer swarm.Address, amount, checkAllo
 	}
 
 	currentTime := s.timeNow().Unix()
-	if currentTime == lastTime.CheckTimestamp {
+	if currentTime <= lastTime.CheckTimestamp || currentTime <= lastTime.Timestamp {
```

Comment:
Would be really nice to have an elaborate comment on this line, since it is so important